src/xdp/program.c

```diff
@@ -1633,7 +1633,10 @@ EbpfProgramOnClientAttach(
     TraceEnter(
         TRACE_CORE, "AttachingProvider=%p AttachingClient=%p", AttachingProvider, AttachingClient);
 
-    if (ClientData == NULL || ClientData->header.size != sizeof(IfIndex) || ClientData->data == NULL) {
+    if (ClientData == NULL ||
+        ClientData->header.version != 1 ||
+        ClientData->header.size != sizeof(IfIndex) ||
```

Comment:
This is why I don't like having both a size and version. Effectively you get two version numbers you have to compare.

Reply:
the way eBPF does it has its upsides: the structs can be extended without making a breaking change (by changing only the size field). breaking changes can be implemented by changing the version field.

doing it the "one version field" way, like XDP, requires introducing an entirely different type of struct if you must make a breaking change.